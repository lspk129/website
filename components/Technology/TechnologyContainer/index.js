import styled from 'styled-components';

const TechnologyContainer = styled.div`
  text-align: center;
  display: grid;
  grid-template-columns: 375px 375px;
  grid-template-rows: 260px;
  padding-bottom: 100px;
  justify-content: center;
  align-content: center;
`;

export default TechnologyContainer;
