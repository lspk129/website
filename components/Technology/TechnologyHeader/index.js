import styled from 'styled-components';

const TechnologyHeader = styled.h2`
  text-align: center;
  padding: 60px;
`;

export default TechnologyHeader;
