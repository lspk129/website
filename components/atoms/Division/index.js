import styled from 'styled-components';

const Division = styled.div`
  ${props => props.textAlign && `text-align: ${props.textAlign}`};
  // margin properties
  ${props => props.m && `margin: ${props.m}rem;`}
  ${props => props.mx && `margin: 0rem ${props.mx}rem;`}
  ${props => props.my && `margin: ${props.my}rem 0rem;`}
  ${props => props.mt && `margin-top: ${props.mt}rem;`}
  ${props => props.mb && `margin-bottom: ${props.mb}rem;`}
  ${props => props.mr && `margin-right: ${props.mr}rem;`}
  ${props => props.ml && `margin-left: ${props.ml}rem;`}
  // padding properties
  ${props => props.p && `padding: ${props.p}rem;`}
  ${props => props.px && `padding: 0rem ${props.px}rem;`}
  ${props => props.py && `padding: ${props.py}rem 0rem;`}
  ${props => props.pt && `padding-top: ${props.pt}rem;`}
  ${props => props.pb && `padding-bottom: ${props.pb}rem;`}
  ${props => props.pr && `padding-right: ${props.pr}rem;`}
  ${props => props.pl && `padding-left: ${props.pl}rem;`}
  // background properties
  ${props => props.bg && `
    background-image: url(${props.src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
`;

export default Division;
