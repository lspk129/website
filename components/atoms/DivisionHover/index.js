import styled from 'styled-components';

const DivisionHover = styled.div`
  transition: all 0.5s ease-in-out;
  ${props => props.height && `height: ${props.height}px`};
  ${props => props.scale && `
    img {
      width: 200px;
      height: 200px;
    }
    &:hover {
    transform: scale(1.1);
  }

  `}
  ${props => props.bgcolor && `
    &:hover {
      background: ${props.theme.background};
      color: ${props.theme.text};
      .icons {
        color: ${props.theme.text};
      }
    };
    .icons {
      transition: all 0.5s ease-in-out;
      color: ${props.theme.background};
    }
  `}
`;

export default DivisionHover;
