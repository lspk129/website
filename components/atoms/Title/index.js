import React from 'react';
import styled from 'styled-components';

const Title = styled(({ tag, children, ...props }) => React.createElement(tag, props, children))`
  text-align: center;

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ tag, upperCase }) => tag === 'h3' && `
    ${upperCase && 'text-transform: uppercase'}
  `}
`;

Title.defaultProps = {
  tag: 'h1',
};

export default Title;
