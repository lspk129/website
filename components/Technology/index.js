import React from 'react';

import TechnologyHeader from './TechnologyHeader';
import TechnologyContainer from './TechnologyContainer';
import TechnologyItem from './TechnologyItem';

const Technology = () => (
  <div>
    <TechnologyHeader>
      We use cutting-edge technologies
    </TechnologyHeader>
    <TechnologyContainer>
      <TechnologyItem>
        <img src="/static/React_logo.png" alt="React" />
        <h3>React</h3>
      </TechnologyItem>
      <TechnologyItem>
        <img src="/static/Meteor_logo.png" alt="Meteor" />
        <h3>Meteor</h3>
      </TechnologyItem>
      <TechnologyItem>
        <img src="/static/Nextjs_logo.png" alt="Next" />
        <h3>Next.js</h3>
      </TechnologyItem>
    </TechnologyContainer>
  </div>
);

export default Technology;
