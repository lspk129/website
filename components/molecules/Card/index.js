import React from 'react';
import PropTypes from 'prop-types';

import Division from '../../atoms/Division';
import Title from '../../atoms/Title';

const Card = ({ image, label, text }) => (
  <Division textAlign={'center'} py={3} pr={1} pl={1}>
    {image}
    <Title tag={'h3'}>{label}</Title>
    <Title tag={'p'}>{text}</Title>
  </Division>
);

Card.propTypes = {
  image: PropTypes.node,
  label: PropTypes.string,
  text: PropTypes.string,
};

Card.defaultProps = {
  image: '',
  label: '',
  text: '',
};

export default Card;
