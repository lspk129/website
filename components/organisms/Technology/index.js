import React from 'react';
import { Row, Col } from 'react-styled-flexboxgrid';

import Card from '../../molecules/Card';
import Subtitle from '../../atoms/Subtitle';
import Division from '../../atoms/Division';
import DivisionHover from '../../atoms/DivisionHover';

const Technology = () => (
  <Division pb={3}>
    <Division py={3}>
      <Subtitle>
        We use cutting-edge technologies
      </Subtitle>
    </Division>
    <Row center={'xs'}>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale>
          <Card
            image={<img src="/static/React_logo.png" alt="React" />}
            label={'React'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale>
          <Card
            image={<img src="/static/Meteor_logo.png" alt="Meteor" />}
            label={'Meteor'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale>
          <Card
            image={<img src="/static/Nextjs_logo.png" alt="Next.js" />}
            label={'Next.js'}
          />
        </DivisionHover>
      </Col>
    </Row>
  </Division>
);

export default Technology;
