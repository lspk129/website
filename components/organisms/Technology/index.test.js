import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Technology from './';
import Card from '../../molecules/Card';
import Title from '../../atoms/Title';
import Division from '../../atoms/Division';
import DivisionHover from '../../atoms/DivisionHover';

describe('Technology ', () => {
  const mountWrapper = mount(<Technology />);

  it('renders correctly', () => {
    const wrapper = renderer.create(<Technology />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Title', () => {
    expect(mountWrapper).toBePresent();
    expect(mountWrapper.find(Title)).toBePresent();
  });

  it('renders Title component with content "We use cutting-edge technologies"', () => {
    expect(mountWrapper.find(Title).at(0)).toHaveText('We use cutting-edge technologies');
  });

  it('renders DivisionHover component', () => {
    expect(mountWrapper.find(DivisionHover)).toBePresent();
  });

  it('renders Division component', () => {
    expect(mountWrapper.find(Division)).toBePresent();
  });

  it('renders Card component with label "React"', () => {
    expect(mountWrapper.find(Card).at(0)).toHaveProp('label', 'React');
  });

  it('renders Card component with label "Meteor"', () => {
    expect(mountWrapper.find(Card).at(1)).toHaveProp('label', 'Meteor');
  });

  it('renders Card component with label "Next.js"', () => {
    expect(mountWrapper.find(Card).at(2)).toHaveProp('label', 'Next.js');
  });

  it('renders img tag to the technologies section', () => {
    expect(mountWrapper.find('img').at(0)).toHaveProp('src', '/static/React_logo.png');
  });

  it('renders img tag to the technologies section', () => {
    expect(mountWrapper.find('img').at(1)).toHaveProp('src', '/static/Meteor_logo.png');
  });


  it('renders img tag to the technologies section', () => {
    expect(mountWrapper.find('img').at(2)).toHaveProp('src', '/static/Nextjs_logo.png');
  });
});
