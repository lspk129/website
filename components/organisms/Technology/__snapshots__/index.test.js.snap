// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technology  renders correctly 1`] = `
<div
  className="sc-bdVaJa ePJfYc"
>
  <h2
    className="sc-bxivhb kixuJS"
  >
    We use cutting-edge technologies
  </h2>
  <div
    className="Row__Row-kNtYPA jaQNpG"
  >
    <div
      className="Col__Col-hCQanm fyLqBb"
    >
      <div
        className="sc-ifAKCX haIUXD"
        scale={true}
      >
        <div
          className="sc-bdVaJa iGScBl"
        >
          <img
            alt="React"
            src="/static/React_logo.png"
          />
          <h3
            className="sc-bwzfXH gCrSZp"
          >
            React
          </h3>
          <p
            className="sc-htpNat KtFsv"
          >
            
          </p>
        </div>
      </div>
    </div>
    <div
      className="Col__Col-hCQanm fyLqBb"
    >
      <div
        className="sc-ifAKCX haIUXD"
        scale={true}
      >
        <div
          className="sc-bdVaJa iGScBl"
        >
          <img
            alt="Meteor"
            src="/static/Meteor_logo.png"
          />
          <h3
            className="sc-bwzfXH gCrSZp"
          >
            Meteor
          </h3>
          <p
            className="sc-htpNat KtFsv"
          >
            
          </p>
        </div>
      </div>
    </div>
    <div
      className="Col__Col-hCQanm fyLqBb"
    >
      <div
        className="sc-ifAKCX haIUXD"
        scale={true}
      >
        <div
          className="sc-bdVaJa iGScBl"
        >
          <img
            alt="Next.js"
            src="/static/Nextjs_logo.png"
          />
          <h3
            className="sc-bwzfXH gCrSZp"
          >
            Next.js
          </h3>
          <p
            className="sc-htpNat KtFsv"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
